src/jit/optimizer.cpp

```diff
@@ -7962,7 +7962,11 @@ void                Compiler::optOptimizeBools()
                 continue;
             if (genTypeSize(t1->TypeGet()) != genTypeSize(t2->TypeGet()))
                 continue;
-
+#ifdef _TARGET_ARMARCH_
+            // Skip the small operand which we cannot encode.
+            if (genTypeSize(c1->TypeGet()) < 4)
+                continue;
```

Comment:
Instead of comparing with 4 it is preferred to use this:
varTypeIsSmall(c1->TypeGet())